resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -922,7 +881,10 @@ private Future<List<DnsRecord>> resolveAll(DnsQuestion question, DnsRecord[] add
         // It was not A/AAAA question or there was no entry in /etc/hosts.
         final DnsServerAddressStream nameServerAddrs =
                 dnsServerAddressStreamProvider.nameServerAddressStream(hostname);
-        new DnsRecordResolveContext(this, ch, channelReadyPromise, promise, question, additionals,
+        ChannelFuture resolveChannelFuture = resolveChannelProvider.nextResolveChannel();
+        promise.addListener(new DnsResolveChannelFutureListener(resolveChannelProvider, resolveChannelFuture));
```

Comment:
I think in this case you want a listener that will only do smth on failure. On success we should not close the channel (in case of `DnsResolveNewChannelProvider`).

Reply:
Isn’t that what we do in the different impl ?

Reply:
oh, nvm, you attach this listener to `promise`, I thought it's for `resolveChannelProvider`